## Final Project Submission

Please fill out:
* Student name: 
* Student pace: self paced / part time / full time
* Scheduled project review date/time: 
* Instructor name: 
* Blog post URL:


In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib notebook
import seaborn as sns

An attempt to write a for loop to process files given a list|Dict of names

In [2]:
# Your code here - remember to use markdown cells for comments as well!

# datasets_dict = {'bom.movie_gross':'bom_movie_gross','imdb.name.basics':'imdb_name_basics',
#             'imdb.title.akas':'imdb_title_akas','imdb.title.basics':'imdb_title_basics',
                
#            'imdb.title.crew':'imdb_title_crew','imdb.title.principals':'imdb_title_principals',
#             'imdb.title.ratings':'imdb_title_ratings','tmdb.movies':'tmdb_movies',
#            'tn.movie_budgets':'tn_movie_budgets'}

# datasets_csv = ['bom.movie_gross.csv.gz', 'imdb.name.basics.csv.gz', 'imdb.title.akas.csv.gz', 'imdb.title.basics.csv.gz',
#                'imdb.title.crew.csv.gz', 'imdb.title.principals.csv.gz', 'imdb.title.ratings.csv.gz', 'tmdb.movies.csv.gz',
#                'tn.movie_budgets.csv.gz']

# for filename in datasets_csv:
#     data = pd.read_csv(f'zippedData/{filename}', compression='gzip', header=0, sep=',', quotechar='"')
    

# def import_datasets(datasets):
#     for key, value in datasets.items():
#         value = pd.read_csv(f'zippedData/{key}.csv.gz', compression='gzip', 
#                                  header=0, sep=',', quotechar='"')
#         return value

### Import the datasets

These are the _9_ `CSV` datasets

In [3]:
bom_movie_gross = pd.read_csv('zippedData/bom.movie_gross.csv.gz', compression='gzip', 
                                 header=0, sep=',', quotechar='"')
imdb_name_basics = pd.read_csv('zippedData/imdb.name.basics.csv.gz', compression='gzip', 
                                 header=0, sep=',', quotechar='"')
imdb_title_akas = pd.read_csv('zippedData/imdb.title.akas.csv.gz', compression='gzip', 
                                 header=0, sep=',', quotechar='"')
imdb_title_basics = pd.read_csv('zippedData/imdb.title.basics.csv.gz', compression='gzip', 
                                 header=0, sep=',', quotechar='"')
imdb_title_crew = pd.read_csv('zippedData/imdb.title.crew.csv.gz', compression='gzip', 
                                 header=0, sep=',', quotechar='"')
imdb_title_principals = pd.read_csv('zippedData/imdb.title.principals.csv.gz', compression='gzip', 
                                 header=0, sep=',', quotechar='"')
imdb_title_ratings = pd.read_csv('zippedData/imdb.title.ratings.csv.gz', compression='gzip', 
                                 header=0, sep=',', quotechar='"')
tmdb_movies = pd.read_csv('zippedData/tmdb.movies.csv.gz', compression='gzip', 
                                 header=0, sep=',', quotechar='"')
tn_movie_budgets = pd.read_csv('zippedData/tn.movie_budgets.csv.gz', compression='gzip', 
                                 header=0, sep=',', quotechar='"')

These are the _2_ `TSV` datasets

In [4]:
rt_movie_info = pd.read_csv('zippedData/rt.movie_info.tsv.gz', compression='gzip', 
                                 header=0, sep='\t', encoding='ISO-8859-1', quotechar='"')
rt_reviews = pd.read_csv('zippedData/rt.reviews.tsv.gz', compression='gzip', 
                                 header=0, sep='\t', encoding='ISO-8859-1', quotechar='"')

In [5]:
bom_movie_gross.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

### Info 
#### Missing Values
* ~ 1300 missing in Foreign Gross and convert to float
* ~ 30 in domestic gross (already float)
* Studio is missing 5 values, may not be important

In [6]:
bom_movie_gross.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
title             3387 non-null object
studio            3382 non-null object
domestic_gross    3359 non-null float64
foreign_gross     2037 non-null object
year              3387 non-null int64
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


Set index to _movie title_

In [7]:
bom_movie_gross.set_index(['title'], inplace=True)

__Domestic Gross__ 
_Top 5_

Divide column by 1 million to makes totals easier to evaluate

In [8]:
bom_movie_gross['domestic_gross_mm'] = bom_movie_gross['domestic_gross'] / 1000000

In [9]:
bom_movie_gross.sort_values(by='domestic_gross_mm', ascending=False)[:5]

studio  domestic_gross foreign_gross  year  \
title                                                                     
Star Wars: The Force Awakens     BV     936700000.0       1,131.6  2015   
Black Panther                    BV     700100000.0     646900000  2018   
Avengers: Infinity War           BV     678800000.0       1,369.5  2018   
Jurassic World                 Uni.     652300000.0       1,019.4  2015   
Marvel's The Avengers            BV     623400000.0     895500000  2012   

                              domestic_gross_mm  
title                                            
Star Wars: The Force Awakens              936.7  
Black Panther                             700.1  
Avengers: Infinity War                    678.8  
Jurassic World                            652.3  
Marvel's The Avengers                     623.4

If over 1 billion, total is annotated as such `"1,131.6"` see foreign gross for Star Wars from [Box Office Mojo](https://www.boxofficemojo.com/title/tt2488496/?ref_=bo_se_r_8).

__Foreign Gross__

_Cannot sort because dtype is object_

Additionally, scale needs to be adjusted, to the same way `domestic_gross` is annotated, so that data can be analyzed on the same scale. 

In [10]:
bom_movie_gross.sort_values(by='foreign_gross', ascending=False)[:5]

studio  domestic_gross foreign_gross  year  domestic_gross_mm
title                                                                         
Life's a Breeze   Magn.          3500.0         99700  2014             0.0035
The East           FoxS       2300000.0         99700  2013             2.3000
White Lion        Scre.             NaN         99600  2010                NaN
A United Kingdom   FoxS       3900000.0       9900000  2017             3.9000
Le Havre           Jan.        612000.0       9900000  2011             0.6120

_These are the 5 films that had foreign gross > 1 billion_

In [11]:
# filter foreign gross over 1 billion
bill_filt = bom_movie_gross['foreign_gross'].str.contains(',', na=False)
bom_movie_gross[bill_filt]

studio  domestic_gross foreign_gross  year  \
title                                                                     
Star Wars: The Force Awakens     BV     936700000.0       1,131.6  2015   
Jurassic World                 Uni.     652300000.0       1,019.4  2015   
Furious 7                      Uni.     353000000.0       1,163.0  2015   
The Fate of the Furious        Uni.     226000000.0       1,010.0  2017   
Avengers: Infinity War           BV     678800000.0       1,369.5  2018   

                              domestic_gross_mm  
title                                            
Star Wars: The Force Awakens              936.7  
Jurassic World                            652.3  
Furious 7                                 353.0  
The Fate of the Furious                   226.0  
Avengers: Infinity War                    678.8

In [12]:
bom_movie_gross['foreign_gross'].apply(type).value_counts()

<class 'str'>      2037
<class 'float'>    1350
Name: foreign_gross, dtype: int64

#### Create Python Script
* Checks if value is a string 
* If True, check for comma. 
* If True remove comma and convert to float. 
* If False, return float and divide by 1 million 
* Otherwise the value is numeric.

Once column is converted to floats all in the same decimal level, then they can be divided by 1 million to get correct scalling of foreign_gross. 

This function could be called clean_convert_gross.

Add a bool column if `foreign_gross` has a total over 1 billion

In [13]:
from clean_data import clean_convert_gross

In [14]:
bom_movie_gross['foreign_gross_mm'] = bom_movie_gross['foreign_gross'].apply(clean_convert_gross)

In [15]:
# if function messes up drop new column
# bom_movie_gross.drop(labels='foreign_gross_mm', axis=1, inplace=True)

In [16]:
bom_movie_gross.head()

studio  domestic_gross  \
title                                                                
Toy Story 3                                     BV     415000000.0   
Alice in Wonderland (2010)                      BV     334200000.0   
Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
Inception                                       WB     292600000.0   
Shrek Forever After                           P/DW     238700000.0   

                                            foreign_gross  year  \
title                                                             
Toy Story 3                                     652000000  2010   
Alice in Wonderland (2010)                      691300000  2010   
Harry Potter and the Deathly Hallows Part 1     664300000  2010   
Inception                                       535700000  2010   
Shrek Forever After                             513900000  2010   

                                             domestic_gross_mm  \
title                                                            
Toy Story 3                                              415.0   
Alice in Wonderland (2010)                               334.2   
Harry Potter and the Deathly Hallows Part 1              296.0   
Inception                                                292.6   
Shrek Forever After                                      238.7   

                                             foreign_gross_mm  
title                                                          
Toy Story 3                                             652.0  
Alice in Wonderland (2010)                              691.3  
Harry Potter and the Deathly Hallows Part 1             664.3  
Inception                                               535.7  
Shrek Forever After                                     513.9

Sort by new column `foreign_gross_mm` to get top 5 foreign gross.

In [17]:
bom_movie_gross.sort_values(by='foreign_gross_mm', ascending=False)[:5]

studio  domestic_gross foreign_gross  year  \
title                                                                     
Avengers: Infinity War           BV     678800000.0       1,369.5  2018   
Furious 7                      Uni.     353000000.0       1,163.0  2015   
Star Wars: The Force Awakens     BV     936700000.0       1,131.6  2015   
Jurassic World                 Uni.     652300000.0       1,019.4  2015   
The Fate of the Furious        Uni.     226000000.0       1,010.0  2017   

                              domestic_gross_mm  foreign_gross_mm  
title                                                              
Avengers: Infinity War                    678.8            1369.5  
Furious 7                                 353.0            1163.0  
Star Wars: The Force Awakens              936.7            1131.6  
Jurassic World                            652.3            1019.4  
The Fate of the Furious                   226.0            1010.0

Create a `total_gross` column

In [19]:
bom_movie_gross['total_gross'] = bom_movie_gross['domestic_gross_mm'] + bom_movie_gross['foreign_gross_mm']

In [23]:
bom_movie_gross.sort_values(by='total_gross', ascending=False)[:5]

studio  domestic_gross foreign_gross  year  \
title                                                                     
Star Wars: The Force Awakens     BV     936700000.0       1,131.6  2015   
Avengers: Infinity War           BV     678800000.0       1,369.5  2018   
Jurassic World                 Uni.     652300000.0       1,019.4  2015   
Marvel's The Avengers            BV     623400000.0     895500000  2012   
Furious 7                      Uni.     353000000.0       1,163.0  2015   

                              domestic_gross_mm  foreign_gross_mm  total_gross  
title                                                                           
Star Wars: The Force Awakens              936.7            1131.6       2068.3  
Avengers: Infinity War                    678.8            1369.5       2048.3  
Jurassic World                            652.3            1019.4       1671.7  
Marvel's The Avengers                     623.4             895.5       1518.9  
Furious 7                                 353.0            1163.0       1516.0